### PR TITLE
add whitelist library

### LINKED ISSUE #166: Add app engine standard whitelisting to the library
Currently both Eclipse and Intellij require this.

See https://github.com/GoogleCloudPlatform/gcloud-intellij/issues/784#issuecomment-232531584

### PR DESCRIPTION
**Pull request by elharo**
@etanshaul @joaoandremartins fix #166

### CODE REVIEW

**Comment by etanshaul** [comment]
I forwarded you by email the link to the internal whitelist that is used. It looks a little different than this. I wonder if we want to sync it up?

**Comment by elharo** [comment]
This is based on that. The plan overall is to deprecate the internal one
though.

On Thu, Jul 14, 2016 at 1:23 PM, Etan Shaul <notifications@github.com>
wrote:

> I forwarded you by email the link to the internal whitelist that is used.
> It looks a little different than this. I wonder if we want to sync it up?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/167#issuecomment-232733188>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA9X6CaklmZy1NOWCBUtNSxngjoDdpKdks5qVnCdgaJpZM4JMqGo>
> .
>



-- 
Elliotte Rusty Harold
elharo@macfaq.com

**Comment by joaoandremartins** [comment]
I didn't check if the list matched the one in our docs. Checkstyle is failing because some lines are too big. Other than that, lgtm.

**Comment by etanshaul** [comment]
lgtm to me too. but maybe @loosebazooka or @meltsufin will want to check to make sure it fits in with the design

**Comment by loosebazooka** [comment]
just out of curiosity, isn't this in the appengine sdk somewhere?

**Comment by etanshaul** [comment]
Its here https://mvnrepository.com/artifact/com.google.appengine/appengine-tools-sdk/1.9.38

But is this something we'd want to use?

**Comment by loosebazooka** [comment]
I see apphosting.runtime.security.WhiteList in appengine-tools-api.jar, which we can get from the cloudsdk object. It does make the whitelist check a little more difficult though.

File: src/main/java/com/google/cloud/tools/appengine/api/whitelist/AppEngineJreWhitelist.java [hunk 1]
```diff
@@ -0,0 +1,1979 @@
...
+              "javax.script.SimpleScriptContext",
+              "javax.sql.CommonDataSource",
+              "javax.sql.ConnectionEvent",
+              "javax.sql.ConnectionEventListener",
+              "javax.sql.ConnectionPoolDataSource",
+              "javax.sql.DataSource",
+              "javax.sql.PooledConnection",
+              "javax.sql.RowSet",
+              "javax.sql.RowSetEvent",
+
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/api/whitelist/AppEngineJreWhitelist.java]: I'm confused what's going on here. Are these not covered by the whitelist itself?

**Comment by elharo** [comment]
I don't see it in appengine-tools-api.jar,

$ jar tvf appengine-tools-api.jar|grep whitelist
elharo-macbookair2:Downloads elharo$

**Comment by patflynn** [comment]
I imagine there was an offline conversation here. Why did appu's tools
include the whitelist and Elliotte's not?

**Pull request merged**